lib/diaspora/shareable.rb

```diff
@@ -55,9 +57,14 @@ def self.owned_by_user(user)
           user.person.send(table_name).where(pending: false)
         end
 
+        def self.shareable_initialize(params)
+          new(params.to_hash.stringify_keys.slice(*column_names)).tap do |new_post|
```

Comment:
I can still change it to `shareable` if you want :)

The greatest coding problem ever!